qa/index.js

```diff
@@ -655,23 +533,16 @@ function colorizeWorker() {
         case 'load':
           const timeStart = performance.now();
           self.postMessage({ type: 'loading', ...data });
-          doc = libreoffice.loadDocumentFromArrayBuffer(data.data);
+          doc = await libreoffice.loadDocumentFromArrayBuffer(data.data);
           console.log(`Loaded document in ${performance.now() - timeStart}ms`);
           self.postMessage({ type: 'loaded', file: data.file });
-          const xDoc = doc;
-          xDoc.startBatchUpdate();
-          const text = xDoc.getText();
-          colorize(text, shouldStop);
-          // Skip calling finishBatchUpdate because the document will be saved as a PDF and discarded
-          xDoc.as('frame.XStorable').storeToURL(data.file + '.pdf', {
-            FilterName: 'writer_pdf_Export',
-          });
-          xDoc.as('util.XCloseable').close(true);
+          doc.postUnoCommand('.uno:Colorize');
+          await doc.saveAs(data.file + '.pdf', 'pdf');
           self.postMessage({
             type: 'finished',
             time: (performance.now() - timeStart) / 1000,
           });
-          self.close();
+          // self.close();
```

Comment:
```suggestion
```

Reply:
colorizeWorker itself should be removed since worker support has been removed as well, but I'll keep it there for reference